org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/sourcelookup/advanced/JDIHelpers.java

```diff
@@ -70,10 +71,10 @@ public File getClassesLocation(Object element) throws DebugException {
 			try {
 				URL url = new URL(locations[1]);
 				if ("file".equals(url.getProtocol())) { //$NON-NLS-1$
-					return new File(url.getPath()).toPath().normalize().toFile();
+					return new File(url.toURI()).toPath().normalize().toFile();
```

Comment:
Just an observation: in `org.eclipse.core.runtime.URIUtil.toFile(URI)` `new File(uri.getSchemeSpecificPart())` is used.

Also `Path.normalize()` seem to be unneeded with `File(URI)` constructor.